runtime/vm/UpcallVMHelpers.cpp

```diff
@@ -292,12 +292,14 @@ native2InterpJavaUpcallImpl(J9UpcallMetaData *data, void *argsListPointer)
 				I_64 argValue = *(I_64*)getArgPointer(nativeSig, argsListPointer, argIndex);
 #if !defined(J9VM_ENV_LITTLE_ENDIAN)
 				/* Right shift the 64-bit float argument by 4 bytes(32 bits) given the actual value
-				 * is placed on the higher 4 bytes on the Big-Endian(BE) platforms.
+				 * is placed on the higher 4 bytes on the Big-Endian(BE) platforms after being converted
+				 * to long via floatToLongArg() in the downcall handler.
+				 * See InternalDowncallHandler.java for details.
 				 */
 				if (J9_FFI_UPCALL_SIG_TYPE_FLOAT == argSigType) {
 					argValue = argValue >> J9_FFI_UPCALL_SIG_TYPE_32_BIT;
 				}
```

Comment:
I think the shift is required for all 32-bit types (_except_ perhaps float). A 64-bit value was read above on line 292. If only the first 32-bits are wanted, they are the top 32 bits of are what is needed on a big-endian platform unless, in the case of float, the value was first extended to 64 bits.

Reply:
I just removed the check on `float` and will need to verify on the supported platforms (specially on AIX and zLinux) to see whether it works good as expected.

Reply:
The shift doesn't work for other 32bit types on the Big-Endian platforms (e.g. AIX or zLinux) except float in that float is passed in the lower address 4-byte of the slot, while char/short/int are effectively sign-extended to int64 before passing (so they are treated as if they were int64 (confirmed by @zl-wang), which follows what is passed in interpreter.  So we still need to type check for `float` to do shift.

Reply:
I just restored the type check for `float` in such  case.